build_all.sh

```diff
@@ -8,30 +9,32 @@ if [ -z "$XGBOOST_VERSION" ]; then
     exit 1
 fi
 
-# install gpuci tools
-curl -s https://raw.githubusercontent.com/rapidsai/gpuci-tools/main/install.sh | bash
-
 . /opt/conda/etc/profile.d/conda.sh
 conda activate rapids
 
 # Install `boa` for `mambabuild`
-conda install -yq boa
+mamba install -y boa
```

Comment:
Wonder if we should move this to the `base` environment. Think that is where `conda` would look when running `conda mambabuild`

Reply:
> Started writing a review, but wasn't fast enough sweat_smile

Whoops! We can always revise if needed. This has worked in the past, so I'm not sure if that would be the issue. Eventually we'll switch `xgboost` to use GH Actions and therefore our new CI images, which has all of the `mamba`/`boa` stuff included correctly in the `base` environment.

Also, when I was debugging locally, I did try to use our new GH Action CI images and the issue still persisted. I'm really not sure what's causing this problem :thinking:

Reply:
No worries 🙂

Yeah I'm not sure either. Normally would expect these things in `base`. Though IIUC we are moving everything to `base`, so this may be moot.

Interesting do you have the output? Or is it just the same as the Jenkins build?

Reply:
My local output was the same as the Jenkins build. It seems that `conda` was complaining about not being able to find `py-xgboost` before it actually built `py-xgboost`.

And yes, for the GH Actions images, we put everything in the `base` environment. The Jenkins images aren't as tidy, but they'll be going away soon anyway.